src/Functions/indexHint.cpp

```diff
@@ -39,6 +39,8 @@ class FunctionIndexHint : public IFunction
 
     bool useDefaultImplementationForNulls() const override { return false; }
 
+    bool useDefaultImplementationForNothing() const override { return false; }
```

Comment:
why?

Reply:
Added it while experimenting, forgot to remove. Fixed